### PR TITLE
Feat:[UIUX-45]notfications enhanced filters

### PR DESCRIPTION
**Pull request by sidharthjain-zemoso**
# Description
1. Integrated Dynamic and enhanced filters in notifications
2. Modified logic to add calendar as filter
3. Clear sort when screen breaks because of sort.
4. Corrected date format in filter chip

Corrected date shown in filter chip (earlier it was showing according to UTC):
image

Multivalue filters:
image

Dynamic filters:
image

image
image

Dynamic Calendar:
image

Corrected date format in filter chip:
image

Please include a summary of the changes and the related issue. Please also include relevant motivation and context. List
any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also
list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] The commit message/PR follows our guidelines

# **Other information**:

### CODE REVIEW

**Review by mkinchin-paladin — CHANGES_REQUESTED** [review]
Is there a ticket for these changes? Looking for description 
I think date format in the table and in the filter should be the same.
Also would be nice to see "Before" and "After" if ticket lacks detailed description on how it should look like

**Comment by sidharthjain-zemoso** [comment]
> Is there a ticket for these changes? Looking for description I think date format in the table and in the filter should be the same. Also would be nice to see "Before" and "After" if ticket lacks detailed description on how it should look like

This is a ported PR from SaaS. 
The filter here is a calendar. UI changes the selected date to UTC and this UTC time was being used while sending request to backend API. This conversion was some how was bound to a variable which was causing an issue. 
Eg: when user selected date range 2023-11-7 to 2023-11-8, UI was showing it as 2023-11-6 18:30:00 - 2023-11-8 18:29:00.

Also, the current calendar filter doesn't support time selection.

**Review by mkinchin-paladin — APPROVED** [review]

**Review by mkinchin-paladin — CHANGES_REQUESTED** [review]
Had a chat with @plyubich - we must have a standardize date format across whole UI. (and we had it before).
Now Date format in filter and date format in column are different. Pls make sure to use global date format settings in filters.

**Comment by sidharthjain-zemoso** [comment]
> Also please truncate global date format to seconds, no need in more granular time.

It has remained unchanged since the calendar filter was added. Can we please handle that in new ticket?

**Comment by plyubich** [comment]
@sidharthjain-zemoso this is not fixed. The display format are different between list view and filter. Why? I thought we have a common date format that is used globally in the UI.

Cc: @JaysinhGo 

image

**Review by plyubich — CHANGES_REQUESTED** [review]
Date on the UI need to use the same global format.

**Comment by sidharthjain-zemoso** [comment]
> @sidharthjain-zemoso this is not fixed. The display format are different between list view and filter. Why? I thought we have a common date format that is used globally in the UI.
> 
> Cc: @JaysinhGo
> 
> image

It was that way from the time of calendar filter was added.
I have made the code changes to fix that. Please have a look:
image

File: webapp/src/app/pacman-features/modules/admin/policies/policies.component.ts [hunk 1]
```diff
@@ -587,6 +587,7 @@
          })["optionValue"];
          this.fieldName = apiColName;
        }catch(e){
+        this.headerColName = '';
```
Comment (mkinchin-paladin) [inline webapp/src/app/pacman-features/modules/admin/policies/policies.component.ts:590]: indentation

File: webapp/src/app/shared/date-selection/date-selection.component.ts [hunk 2]
```diff
@@ -16,16 +16,27 @@ import {
         },
     ],
 })
-export class DateSelectionComponent implements OnInit {
+export class DateSelectionComponent implements OnInit, OnChanges {
     readonly today = new Date();
-
+
```
Comment (mkinchin-paladin) [inline webapp/src/app/shared/date-selection/date-selection.component.ts:21]: remove

File: webapp/src/app/shared/date-selection/date-selection.component.ts [hunk 3]
```diff
@@ -16,16 +16,27 @@ import {
...
-
+    
     @Input("minDate") minDate : Date;
+    @Input() maxDate: Date = this.today;
 
     @Input() selectedRange?: DateRange<Date>;
 
     @Output() datesSelected = new EventEmitter<{ from: Date; to: Date }>();
-
+
```
Comment (mkinchin-paladin) [inline webapp/src/app/shared/date-selection/date-selection.component.ts:28]: remove

File: webapp/src/app/shared/services/filter-management.service.ts [hunk 4]
```diff
@@ -289,7 +289,7 @@
        }
    
        let filterTagsData = await this.getFilterTagsData(payload, currentFilterType.optionURL);
-       filterTagsData = updateFilterTags(filterTagsData, currentFilterType.optionName);
+       if(updateFilterTags) filterTagsData = updateFilterTags(filterTagsData, currentFilterType.optionName);
```
Comment (mkinchin-paladin) [inline webapp/src/app/shared/services/filter-management.service.ts:292]: not sure about TS code style standards but let's have condition and its code on different lines

**Review by mkinchin-paladin — COMMENTED** [review]
Will changes from this [PR](https://github.com/PaladinCloud/EE/pull/1532) affect filters visually?

**Comment by sidharthjain-zemoso** [comment]
> Will changes from this [PR](https://github.com/PaladinCloud/EE/pull/1532) affect filters visually?

We are just changing the rendering of date range filter chip value by adding a date range formatter pipe. This will affect only the calendar filter chip display value and rest all will be same.

File: webapp/src/app/shared/services/filter-management.service.ts (hunk 4 above)
Comment (sidharthjain-zemoso) [reply webapp/src/app/shared/services/filter-management.service.ts]: Done.

File: webapp/src/app/pacman-features/modules/admin/policies/policies.component.ts (hunk 1 above)
Comment (sidharthjain-zemoso) [reply webapp/src/app/pacman-features/modules/admin/policies/policies.component.ts]: Done

File: webapp/src/app/shared/date-selection/date-selection.component.ts (hunk 3 above)
Comment (sidharthjain-zemoso) [reply webapp/src/app/shared/date-selection/date-selection.component.ts]: Done

File: webapp/src/app/shared/date-selection/date-selection.component.ts (hunk 2 above)
Comment (sidharthjain-zemoso) [reply webapp/src/app/shared/date-selection/date-selection.component.ts]: Done

**Review by plyubich — DISMISSED** [review]

**Review by go-jay — APPROVED** [review]

**Review by santhosh-challa — APPROVED** [review]

**Pull request merged**